abn/service_impl.go

```diff
@@ -0,0 +1,147 @@
+package abn
+
+// lookup.go -(internal) implementation of gRPC Lookup method
+
+import (
+	"errors"
+	"fmt"
+	"hash/maphash"
+	"reflect"
+	"strconv"
+	"strings"
+
+	"github.com/google/uuid"
+	"github.com/iter8-tools/iter8/base/log"
+	"github.com/iter8-tools/iter8/controllers"
+)
+
+var allRoutemaps controllers.AllRouteMapsInterface = &controllers.DefaultRoutemaps{}
+
+// versionHasher is hash used for selecting versions
+var versionHasher maphash.Hash
+
+// lookupInternal is detailed implementation of gRPC method Lookup
+// application is a of the form "namespace/name"
+func lookupInternal(application string, user string) (controllers.RoutemapInterface, *int, error) {
```

Comment:
Why is it necessary to return a pointer of an int instead of just the int itself?

Reply:
It probably isn't necessary, I will modify this.